### ISSUE TITLE
Tesoro forced to run as root user

### ISSUE THREAD

**Issue by ademariag** [kapicorp/tesoro#1]
* TL;DR
Although the design of **`tesoro`** would allow it to run as non-root user, we are forced to run as root for a number of reasons.

* Reason
Ideally, we would want to run **`tesoro`** on port `8443` (unprivileged) and then user the service to expose it as port `443`.
This works on both KIND and GKE "non private" clusters.

Unfortunately on private GKE clusters, although the **`apiserver`** is meant to connect to the tesoro webhook using the service IP on port 443, the connection seems to then reach the pod IP directly, expecting tesoro to be running on the same port. 

Logs:

* service configuration
```
$ kubectl get services
NAME                          TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)             AGE
tesoro ClusterIP   10.15.5.59   <none>        443/TCP,9095/TCP   52s
```
* actual pod running on port 8443
```
$ kubectl apply -f example-gkms-secret.yml
Error from server (InternalError): error when creating "example-gkms-secret.yml": Internal error occurred: failed calling webhook "tesoro.kapicorp.com
": Post https://tesoro.tesoro.svc:443/mutate?timeout=30s: no service port '\u20fb' found for service "tesoro"  
```
We've also experienced other kind of issues like

```
$ kubectl apply -f example-gkms-secret.yml
Error from server (InternalError): error when creating "example-gkms-secret.yml": Internal error occurred: failed calling webhook "tesoro.kapicorp.com": Post https://tesoro.tesoro.svc:443/mutate?timeout=5s: dial tcp 10.16.0.18:8443: i/
o timeout
```

while this is probably because of firewall rules: https://gitlab.com/gitlab-org/charts/gitlab/-/issues/1809

```
Error from server (InternalError): error when creating "example-gkms-secret.yml": Internal error occurred: failed calling webhook "tesoro.kapicorp.com": Post https://tesoro.tesoro.svc:443/mutate?timeout=30s: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

Because of this https://github.com/kubernetes/kubernetes/issues/61510 although the api supports it, we cannot run both service and pod on 8443 either.

The only solution that seems to work is to have the actual **`tesoro`** pod also running on port 443, which makes it work instantly :(

**Comment by uberspot** [comment]
Just for reference here btw, running as root is also not the best solution for this. 
https://github.com/moby/moby/issues/8460#issuecomment-87123934
Ideally we'd just add the `cap_net_bind_service=+ep` to the tesoro binary and run that with the CAP privileges added to k8s (?). That would still be a smaller attack surface than full root perms. 
Maybe this could be improved in a further PR.

**Comment by ademariag** [comment]
yes the idea for the upcoming 'kapitan-reference' deployment is to use CAP + readonly filesystem

**Comment by ademariag** [comment]
good the idea to also add it to the binary! @uberspot
